prepline_general/api/general.py

```diff
@@ -233,6 +234,19 @@ def pipeline_api(
             )
         )
     )
+
+    # If this var is set, reject traffic when free memory is below minimum
+    # Allow internal requests - these are parallel calls already in progress
+    mem = psutil.virtual_memory()
+    memory_free_minimum = int(os.environ.get("UNSTRUCTURED_MEMORY_FREE_MINIMUM_MB", 0))
+
+    if memory_free_minimum > 0 and mem.available <= memory_free_minimum * 1024 * 1024:
+        origin = str(request.client.host)
```

Comment:
i think getting origin ip should be similar to this https://github.com/Unstructured-IO/unstructured-infra-tools/blob/main/unstructured_infra_tools/api_requests/api_quota_check.py#L37-L40 since the proxy jump